src/java/detectors/avoid_reset_exception_rule/AvoidResetExceptionRuleNonCompliant.java

```diff
@@ -0,0 +1,23 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+// {fact rule=java-avoid-reset-exception@v1.0 defects=1}
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import lombok.extern.slf4j.Slf4j;
+import java.io.InputStream;
+
+@Slf4j
+public class AvoidResetExceptionRuleNonCompliant {
+
+    // NonCompliant: `PutObjectRequest` created without setting read limit, risking `ResetException`.
```

Comment:
Let's not use backticks for functions / class definitions. Applicable for all files.